lib/checkunusedvar.cpp

```diff
@@ -1586,3 +1589,69 @@ bool CheckUnusedVar::isEmptyType(const Type* type)
     emptyType=false;   // unknown types are assumed to be nonempty
     return emptyType;
 }
+
+bool CheckUnusedVar::isFunctionWithoutSideEffects(const Function& func, const Token& functionUsageToken) {
```

Comment:
I think it's weird to pass a `Token` by reference. Please pass it by pointer instead.